oda_api/api.py

```diff
@@ -153,13 +153,15 @@ def func_wrapper(*args, **kwargs):
                 n_tries_left -= 1
 
                 if n_tries_left > 0:
-                    logger.warning("problem in API call, %i tries left:\n%s\n sleeping %i seconds until retry",
+                    logger.debug("problem in API call, %i tries left:\n%s\n sleeping %i seconds until retry",
                                    n_tries_left, message, retry_sleep_s)
+                    logger.warning("possibly temporary problem in API call, %i tries left, sleeping %i seconds until retry",
+                                   n_tries_left, retry_sleep_s)                                   
                     time.sleep(retry_sleep_s)
-            
-            raise RemoteException(
-                message=message
-            )
+                else:
```

Comment:
with this branch, not much changed I guess right? Or is there an additional purpose?

Reply:
this ensures the retry actually happens. It happned not quite as expected.

Oh, I should add a test for it.

Reply:
test added. 
Note that the change in these lines, moving the `raise`, is actually strongly impacting the retry behavior. It enables it. Before, it was kind of broken - apparently.

Reply:
Yes, now, the more I look at it the more I see it. I never realized before.

Also, I would remove that line (135, `UnexpectedDispatcherStatusCode` case), that message is never used, unless we want to use it when raising an exception